### ISSUE TITLE
[ALM Accelerator - QUESTION] Does ALM accelerator merge Customization in same solution from different developers

### ISSUE THREAD

**Issue by linujohn22** [microsoft/coe-starter-kit#4043]
### What is your question?

Consider Below scenario 

I have solution name " SolutionX". The solution contains **Entity A** and **Entity B.**

The same solution imported into Development **Environment A** and Devlopment **Environment B.**

In **Environment A** , we have changed **Entity A** deployed to Validation/Testing . In **Environment B**, we have changed **Entity B**.

 Will the Changes from **Environment B** will be merged to Changes from **Environment A** ? Please advise .

### What component are you experiencing the issue with?

ALM Accelerator Pipelines

### What solution version are you using?

Sep 2022

**Comment by Ballzer0** [comment]
from my experience, the "solution from environment B" would overwrite the solution from environment A.

The toolkit isnt granular enough to detect changes you do in different environments.

This is fine. but its meant we have had to organize our developers to work on different solutions in the stack. so that we dont overwrite each others changes. 

Based on my experience, if you want 2 developers to work on the same solution, they need to work on the same developer environment.

**Comment by linujohn22** [comment]
if we use same development environment, how can we move particular change from single solution can be deployed to TEST from development environment.  Consider Above scenario, how can we move **Entity A** Changes  moved to **Validation/TEST** from **solutionX**  while **Entity B** changes should stay in Devlopment environment.

**Comment by mikefactorial** [comment]
This is a great conversation. Thanks @linujohn22 for posting and @Ballzer0 for your insights. There's a similiar conversation going on in this thread as we speak (https://github.com/microsoft/coe-starter-kit/issues/4030). 

I think @Ballzer0 has hit the nail on the head as far as this discussion of developer isolation vs shared environment goes. 

However, to answer your original question. Assuming the two makers are working on completely separate work in isolated environments then pushing changes to a common downstream environment as managed, the system would merge the work from two maker's environments. Assuming the changes are additive. 

However, this wouldn't be the case for changes to apps or flows, but tables do merge additive changes nicely as do other changes to Dataverse metadata.

In the spirit of this thread maybe we should "merge" this thread with https://github.com/microsoft/coe-starter-kit/issues/4030 :) but I'll keep it open for now.

**Comment by linujohn22** [comment]
> This is a great conversation. Thanks @linujohn22 for posting and @Ballzer0 for your insights. There's a similiar conversation going on in this thread as we speak (#4030).
> 
> I think @Ballzer0 has hit the nail on the head as far as this discussion of developer isolation vs shared environment goes.
> 
> However, to answer your original question. Assuming the two makers are working on completely separate work in isolated environments then pushing changes to a common downstream environment as managed, the system would merge the work from two maker's environments. Assuming the changes are additive.
> 
> However, this wouldn't be the case for changes to apps or flows, but tables do merge additive changes nicely as do other changes to Dataverse metadata.
> 
> In the spirit of this thread maybe we should "merge" this thread with #4030 :) but I'll keep it open for now.

What can be done for flows and app ?

**Comment by mikefactorial** [comment]
@linujohn22 closing out this thread. Between this thread and #4030 I do think we've exhausted this conversation. The idea of developer isolation and how merge conflicts are handled is fairly well known in the platform. Unfortunately, developer isolation isn't realistically possible today without a lot of overhead and merge conflicts will always be an issue even as the platform gets better at reducing the number of noisy diffs if two devs are changing the same thing at the same time in two different environments. However, in your scenario above you are talking about two different devs working on two different flows and committing their changes. In this case Flow A would not have any changes made by developer B so nothing would be overwritten. This is how code first dev works as well with Git. Git is smart enough to know the history of the branch that developer B created to know that no changes were made to Flow A and therefore the changes made by developer A to Flow A that were merged wouldn't be overwritten with anything from developer B's branch.
